proxy_map.go

```diff
@@ -997,6 +1024,24 @@ func (m *Map) tryRemove(ctx context.Context, key interface{}, timeout int64) (in
 	}
 }
 
+func (m *Map) aggregate(ctx context.Context, req *proto.ClientMessage, decoder func(message *proto.ClientMessage) *serialization.Data) (interface{}, error) {
+	resp, err := m.invokeOnRandomTarget(ctx, req, nil)
+	if err != nil {
+		return nil, err
+	}
+	data := decoder(resp)
+	obj, err := m.convertToObject(data)
+	if err != nil {
+		return nil, err
+	}
+	// if this is a canonicalizing set return a slice of interface[}
+	cs, ok := obj.(*iproxy.AggCanonicalizingSet)
+	if ok {
+		return []interface{}(*cs), nil
+	}
```

Comment:
What happens if we don't do this? I assume the user will receive `AggCanonicalizingSet`. Can't the user use it like `[]interface` already, since it is a type alias for this?

Reply:
`m.convertToObject` returns a `*AggCanonicalizingSet` which must be dereferenced before returning to the user, since `AggCanonicalizingSet` is an alias to `[]interface{}`, not `*AggCanonicalizingSet`. Casting that to `[]interface{}` is not necessary though.